drivers/sensor/mcp9808/mcp9808.c

```diff
@@ -35,6 +35,39 @@ int mcp9808_reg_read(const struct device *dev, uint8_t reg, uint16_t *val)
 	return rc;
 }
 
+int mcp9808_reg_write_16bit(const struct device *dev, uint8_t reg,
+			    uint16_t val)
+{
+	const struct mcp9808_data *data = dev->data;
```

Comment:
Is data struct really 'const'?

Reply:
I believe that in this instance it should be `const`. `i2c_write` requires `const struct device *dev` as first parameter.

Reply:
driver `data` is never `const`, devices are.

Reply:
Updated